apps/rewards/app/utils/web3-utils.js

```diff
@@ -0,0 +1,41 @@
+import { toChecksumAddress } from 'web3-utils'
+
+// Check address equality without checksums
+export function addressesEqual(first, second) {
+  first = first && toChecksumAddress(first)
+  second = second && toChecksumAddress(second)
+  return first === second
+}
+
+export const addressPattern = '(0x)?[0-9a-fA-F]{40}'
+
+/**
+ * Shorten an Ethereum address. `charsLength` allows to change the number of
+ * characters on both sides of the ellipsis.
+ *
+ * Examples:
+ *   shortenAddress('0x19731977931271')    // 0x1973…1271
+ *   shortenAddress('0x19731977931271', 2) // 0x19…71
+ *   shortenAddress('0x197319')            // 0x197319 (already short enough)
+ *
+ * @param {string} address The address to shorten
+ * @param {number} [charsLength=4] The number of characters to change on both sides of the ellipsis
+ * @returns {string} The shortened address
+ */
+export function shortenAddress(address, charsLength = 4) {
```

Comment:
Are we using this in other apps?

Reply:
I think we have our own address-shortening function. This one came with the other aragon functions I was utilizing for the Vault integration